crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -0,0 +1,273 @@
+use tedge_mqtt_ext::MqttMessage;
+use tedge_mqtt_ext::Topic;
+
+use super::error::LogManagementError;
+use super::json::CommandStatus;
+use super::json::LogRequestPayload;
+use super::json::LogResponsePayload;
+use super::LogManagerConfig;
+use super::DEFAULT_PLUGIN_CONFIG_FILE_NAME;
+
+fan_in_message_type!(LogInput[MqttMessage, FsWatchEvent] : Debug);
+fan_in_message_type!(LogOutput[MqttMessage]: Debug);
+
+pub struct LogManagerActor {
+    config: LogManagerConfig,
+    plugin_config: LogPluginConfig,
+    mqtt_publisher: LoggingSender<MqttMessage>,
+    messages: SimpleMessageBox<LogInput, NoMessage>,
+    http_proxy: ClientMessageBox<HttpRequest, HttpResult>,
+}
+
+#[async_trait]
+impl Actor for LogManagerActor {
+    fn name(&self) -> &str {
+        "LogManager"
+    }
+
+    async fn run(&mut self) -> Result<(), RuntimeError> {
+        self.reload_supported_log_types().await?;
+
+        while let Some(event) = self.messages.recv().await {
+            match event {
+                LogInput::MqttMessage(message) => {
+                    self.process_mqtt_message(message).await?;
+                }
+                LogInput::FsWatchEvent(event) => {
+                    self.process_file_watch_events(event).await?;
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+impl LogManagerActor {
+    pub fn new(
+        config: LogManagerConfig,
+        plugin_config: LogPluginConfig,
+        mqtt_publisher: LoggingSender<MqttMessage>,
+        messages: SimpleMessageBox<LogInput, NoMessage>,
+        http_proxy: ClientMessageBox<HttpRequest, HttpResult>,
+    ) -> Self {
+        Self {
+            config,
+            plugin_config,
+            mqtt_publisher,
+            messages,
+            http_proxy,
+        }
+    }
+
+    pub async fn process_mqtt_message(&mut self, message: MqttMessage) -> Result<(), ChannelError> {
+        if self.config.logfile_request_topic.accept(&message) {
+            match LogRequestPayload::try_from(message.clone()) {
+                Ok(request) => {
+                    if (request.status.eq(&CommandStatus::Executing)
+                        || request.status.eq(&CommandStatus::Init))
+                        && !self.config.current_operations.contains(&message.topic.name)
+                    {
+                        info!("Log request received: {request:?}");
+                        self.config
+                            .current_operations
+                            .insert(message.topic.name.clone());
+                        self.handle_logfile_request_operation(&message.topic, &request)
+                            .await?;
+                    }
+                }
+                Err(err) => {
+                    error!("Incorrect log request payload: {}", err);
+                }
+            }
+        } else {
+            error!(
+                "Received unexpected message on topic: {}",
+                message.topic.name
+            );
+        }
```

Comment:
I don't see how this can happen (unless the MQTT actor is buggy). Indeed this actor only subscribes to `self.config.logfile_request_topic`.